labonneboite/common/mapping.py

```diff
@@ -129,21 +145,27 @@ def romes_for_naf(self, naf):
         Rome = namedtuple('Rome', ['code', 'name', 'nafs'])
         return [Rome(rome[0], settings.ROME_DESCRIPTIONS[rome[0]], rome[1]) for rome in romes_for_naf]
 
+    @lru_cache(maxsize=8*1024)  # about 500 rome_codes in current dataset and 5000 in sliced dataset
     def nafs_for_rome(self, rome):
```

Comment:
Sauf erreur de ma part, le `lru_cache` ne fonctionne pas bien sur des méthodes, parce que l'objet `self`est sérialisé. Donc n'importe quel changement d'attribut cause un cache miss.

Reply:
euh alors tu me fais des frayeurs, j'ai pas du tout compris `@lru_cache` comme ça moi...   pour moi il cache le résultat pour chaque combinaison possible de paramètres d'entrée de la méthode

https://docs.python.org/3/library/functools.html

Decorator to wrap a function with a memoizing callable that saves up to the maxsize most recent calls. It can save time when an expensive or I/O bound function is periodically called with the same arguments.

En plus j'ai déjà beaucoup travaillé avec les stats de ce cache pour mesurer les hit/miss et tuner la taille, surtout dans create_index.

Reply:
ah ok j'ai pas tilté!! tu parles de méthode de classe  vs méthodes statiques

Reply:
mmm pas trivial ... le seul moyen d'être sûr est d'examiner les stats hit/miss de ce cache, je vais regarder

Reply:
Hmmm après examen, il se trouve que `@lru_cache` ne sérialise *pas* `self`. Ca signifie que si les attributs de `self` sont modifiés, alors la valeur retournée par le cache n'est pas modifiée : 

```from backports.functools_lru_cache import lru_cache

class Foo(object):
    def __init__(self):
        self.plonk = 1

    @lru_cache(100)
    def pouac(self):
        return self.plonk

f = Foo()
print(f.pouac())
f.plonk = 2
print(f.pouac())```

Ca affiche (roulements de tambours) : 

```1
1```

Donc le cache ne va marcher que si les attributs de l'objet ne sont jamais modifiés. Même si par hasard c'est le cas, c'est vraiment piégeux, donc je propose d'extirper le code à cacher dans une fonction séparée.

Reply:
ah ouais bien vu c'est KO! que du miss :(